Update llama-cli documentation

dspasyuk opened the pull request:
- [x] I have read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md)
- Self-reported review complexity:
  - [X ] Low
  - [ ] Medium
  - [ ] High

Updated llama-cli manual

dspasyuk committed:
Update README.md

ggerganov commented on examples/main/README.md:
```diff
@@ -1,6 +1,6 @@
-# llama.cpp/examples/main
+# llama.cpp/examples/llama-cli
```
These titles currently reflect the path of the example within the `llama.cpp` tree, so `llama.cpp/examples/main` is the correct one

ngxson commented on examples/main/README.md:
````diff
@@ -17,60 +17,68 @@ This example program allows you to use various LLaMA language models in an easy
-```powershell
-llama-cli.exe -m models\7B\ggml-model.bin --prompt "Once upon a time"
+```bash
+./llama-cli -m models/gemma-1.1-7b-it.Q4_K_M.gguf -cnv --ctx_size 0 --chat-template llama3
````
specifying `--chat-template` here is misleading, why would user want to use gemma model with llama3 template?

`--ctx_size` is also not necessary

dspasyuk replied:
ups, I did everything for llama and then realised that we do not have any llama models in ggml repo and replaced everything gemma.

ngxson commented on examples/main/README.md:
````diff
@@ -17,60 +17,68 @@ This example program allows you to use various LLaMA language models in an easy
+```
 
-#### Windows:
+### Windows:
````
I don't think windows should be separated since it's just the same with linux/macos (except for the binary name)

This section should be refactored into just "How to run"

dspasyuk committed:
Update README.md

dspasyuk committed:
Update README.md

fixed llama-cli/main, templates on some cmds added chat template sections and fixed typos in some areas

dspasyuk committed:
Update README.md

dspasyuk committed:
Update README.md

dspasyuk committed:
Update README.md
